src/feature/control/control.c

```diff
@@ -7395,6 +7395,17 @@ control_event_transport_launched(const char *mode, const char *transport_name,
                      mode, transport_name, fmt_addr(addr), port);
 }
 
+/** A pluggable transport called <b>transport_name</b> has emitted a log
+ * message found in <b>message</b>. */
+void
+control_event_transport_log(const char *transport_name, const char *message)
+{
+  send_control_event(EVENT_TRANSPORT_LOG,
+                     "650 TRANSPORT_LOG %s %s\r\n",
+                     transport_name,
+                     message);
```

Comment:
Should this message be quoted or escaped somehow?  If not, we should document that.

Reply:
We've removed TRANSPORT_LOG and it's PT_LOG now. Let's use your K/V parser for this instead. Tracking that in https://trac.torproject.org/projects/tor/ticket/28846